sklearn/metrics/pairwise.py

```diff
@@ -30,6 +30,7 @@
 from ..utils._array_api import (
     _find_matching_floating_dtype,
     _is_numpy_namespace,
+    device,
```

Comment:
I think we could also use `get_namespace_and_device` and substitute it to `xp, _ = get_namespace(X, Y)` below